packages/hash/frontend/src/pages/[shortname]/entities/[entity-uuid].page/entity-editor/types-section/entity-type-update-modal.tsx

```diff
@@ -0,0 +1,92 @@
+import { faArrowRight } from "@fortawesome/free-solid-svg-icons";
+import { Button, FontAwesomeIcon } from "@hashintel/hash-design-system";
+import { Box, Stack, Typography } from "@mui/material";
+
+import { Modal } from "../../../../../../components/modals/modal";
+
+interface EntityTypeUpdateModalProps {
+  open: boolean;
+  onClose: () => void;
+  entityTypeTitle: string;
+  currentVersion: number;
+  newVersion?: number;
```

Comment:
This really shouldn't be optional, because we shouldn't even see the modal if there is no `newVersion` – if someone reuses this component elsewhere, they can forget to pass `newVersion` and see `"vundefined"` in the title.

Can we make the rendering of the modal conditional on their being a `newVersion`? e.g.
```ts
{newVersion ? <EntityTypeUpdateModal /> : null}
```
or does that cause problems?

Reply:
Makes sense, updated it here https://github.com/hashintel/hash/pull/1774/commits/008c1a16325559f53ed9f37dffe515f1027821de